src/mainwindow.cpp

```diff
@@ -120,27 +119,25 @@ bool MainWindow::eventFilter(QObject* /*object*/, QEvent* event)
     return false;
 }
 
-
-void MainWindow::when_ReadingList_toggled(bool state)
+void MainWindow::tabChanged(TabType tabType) 
 {
-    if (state) {
-        mp_ui->sideBar->setCurrentWidget(mp_ui->readinglistbar);
+    QAction *readingList = KiwixApp::instance()->getAction(KiwixApp::ToggleReadingListAction);
+    if (tabType == TabType::SettingsTab) { 
+        mp_ui->sideBar->hide();    
+    } else if(tabType == TabType::LibraryTab) { 
+        mp_ui->sideBar->setCurrentWidget(mp_ui->contentmanagerside);
         mp_ui->sideBar->show();
-    }
-    else {
+    } else if (readingList->isChecked()) { 
+        readingListToggled(true);
+    } else {
         mp_ui->sideBar->hide();
     }
 }
 
-void MainWindow::when_libraryPageDisplayed(bool showed)
+void MainWindow::readingListToggled(bool state)
 {
-    auto app = KiwixApp::instance();
-
-    // When library sidebar appeared, or hidden, reading list is always hidden.
-    app->getAction(KiwixApp::ToggleReadingListAction)->setChecked(false);
-
-    if (showed) {
-        mp_ui->sideBar->setCurrentWidget(mp_ui->contentmanagerside);
+    if (state) {
+        mp_ui->sideBar->setCurrentWidget(mp_ui->readinglistbar);
```

Comment:
I think that the diff will be easier to comprehend if you swap the order of these two functions.